dspace-server-webapp/src/main/java/org/dspace/app/rest/ProcessRestController.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.link.process.ProcessResourceHalLinkFactory;
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.model.ProcessRest;
+import org.dspace.app.rest.model.hateoas.BitstreamResource;
+import org.dspace.app.rest.model.hateoas.ProcessFileWrapperResource;
+import org.dspace.app.rest.repository.ProcessRestRepository;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.authorize.AuthorizeException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/" + ProcessRest.CATEGORY + "/" + ProcessRest.PLURAL_NAME)
+public class ProcessRestController {
+
+    private static final Logger log = LogManager.getLogger();
+
+    @Autowired
+    HalLinkService linkService;
+
+    @Autowired
+    private ProcessRestRepository processRestRepository;
+
+    @Autowired
+    private Utils utils;
+
+    @Autowired
+    private HalLinkService halLinkService;
+
+    @Autowired
+    ProcessResourceHalLinkFactory processResourceHalLinkFactory;
+
+    @RequestMapping(method = RequestMethod.GET, value = "/{processId}/files")
```

Comment:
I'm assuming each of these methods should have `@PreAuthorize` annotations to access restrict files in some way?

Reply:
All methods will need a `@PreAuthorize` annotation once you turn this into a LinkRestRepository.